core/src/main/java/org/apache/iceberg/InheritableMetadataFactory.java

```diff
@@ -46,11 +49,14 @@ static class BaseInheritableMetadata implements InheritableMetadata {
     private final int specId;
     private final long snapshotId;
     private final long sequenceNumber;
+    private final String manifestPath;
 
-    private BaseInheritableMetadata(int specId, long snapshotId, long sequenceNumber) {
+    private BaseInheritableMetadata(
+        int specId, long snapshotId, long sequenceNumber, String manifestPath) {
```

Comment:
Passing through the manifest path through inheritable metadata similar to sequence numbers seemed like the cleanest way

Reply:
I agree. It seems easier than hacking `GenericAvroReader` like we do for positions in manifests.

Reply:
One thing that we could do instead is finish implementing default values. Then we could just construct a schema with the value set by default and the reader would automatically do the right thing.

Reply:
Let's do this for now and re-evaluate default values later.

Reply:
I agree. It's easy to change this approach to the default value implementation down the line since currently the change is modification to a single private constructor.